scipy/stats/_continuous_distns.py

```diff
@@ -4346,8 +4346,15 @@ def _stats(self, a, moments='mvsk'):
         return m1, m2, g1, g2
 
     def _entropy(self, a):
-        return a - (a+1.0) * sc.psi(a) + sc.gammaln(a)
+        a = np.asarray([a])
+        def h1(a):
+            return a - (a+1.0) * sc.psi(a) + sc.gammaln(a)
 
+        def h2(a):
+            return -1.5 * np.log(a) + 0.5 * np.log(2 * np.pi) + 0.5
+
+        h = _lazywhere(a >= 1e8, (a), f=h2, f2=h1)
+        return h
```

Comment:
I am not familiar with these formulas, but nitpicking I would suggest calling these functions something like "regular_formula" and "asymptotic formula" as you see in https://github.com/scipy/scipy/pull/18095/files, for example. Also can you clarify the source of the asymptotic formula for those like me? 🙂 Thanks!

Reply:
I've changed the function names. As for the source of the formula, I took the expansions listed in the linked issue and plugged them in.

Reply:
The employed asymptotic expansions should go in comments or at least the PR description. We did this also for the other distributions (see [here](https://github.com/dschmitz89/scipy/blob/76dca342b63357d0ebe472796d1f0feff861ad66/scipy/stats/_continuous_distns.py#L2735) for example). Otherwise it will be hard to maintain in future.

Reply:
Ok, I've added the formulas as comments.